frontend/src/components/taste/TasteList.vue

```diff
@@ -1,88 +1,80 @@
-                {
-                    id: 2,
-                    name: 'Ice-cream',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 3,
-                    name: 'Live-music',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 4,
-                    name: 'Sport-bar',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 5,
-                    name: 'Lunch',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                },
-                {
-                    id: 6,
-                    name: 'Sushi',
-                    check: false,
-                    isAnimate: false,
-                    isClick: false
-                }
-            ]
+            selectedIds: [],
+            tastesData: []
         };
     },
     methods: {
-        checkTaste(index) {
-            let taste = this.tastes[index];
-            taste.check = !taste.check;
-            taste.isClick = true;
-            taste.isAnimate = true;
+        checkTaste(id) {
+            if (this.tastesData[id] === undefined) {
+                this.tastesData[id] = {
+                    check: false,
+                    isClick: false,
+                    isAnimate: false
+                };
+            }
+            let tasteData = this.tastesData[id];
+            if (this.selectedIds.includes(id)) {
+                tasteData.check = false;
+                this.$store.dispatch('taste/deleteUserTaste', id);
+                this.selectedIds.splice(this.selectedIds.indexOf(id), 1);
+            } else {
+                tasteData.check = true;
+                this.$store.dispatch('taste/addUserTaste', this.tastes.byId[id]);
+                this.selectedIds.push(id);
+            }
+            tasteData.isClick = true;
+            tasteData.isAnimate = true;
             setTimeout(function () {
-                taste.isAnimate = false;
+                tasteData.isAnimate = false;
             }, 200);
+        },
+        isChecked(id) {
+            return this.selectedIds.includes(id);
+        },
+        isAnimated(id) {
+            return this.tastesData !== undefined || this.tastesData[id].isAnimate;
+        },
+        isClicked(id) {
+            return this.tastesData !== undefined || this.tastesData[id].isClick;
         }
-    }
+    },
+    computed: {
+        ...mapState('taste', ['tastes', 'userTastes'])
+    },
+    created() {
+        this.$store.dispatch('taste/fetchTastes');
+        this.$store.dispatch('taste/fetchUserTastes').then(() => {
+            this.selectedIds = Object.keys(this.userTastes.byId);
```

Comment:
could be vuex getter `getMyTastesIds() { ... Object.keys(this.userTastes.byId); }`